### ISSUE TITLE
encumbrances

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#808]
http://arctos.database.museum/Encumbrances.cfm is a mess - make it readable

Lots (most?) encumbrances expire "never" or "when research is completed" or some other vague thing that's never going to be recognized.

Lots of encumbrances are made by people who are no longer around (eg, former collection managers).

RECOMMENDATIONS:
- drop (or keep but ignore?) ENCUMBERING_AGENT_ID; all encumbrances are ultimately the responsibility of the current Curator anyway.
- drop (or keep but ignore) EXPIRATION_EVENT
- make EXPIRATION_DATE NOT NULL, require a date in the somewhat-near-future (<5 years?)
- send email reminders when EXPIRATION_DATE is approaching (eg, year, 6 months, 1 month)
- automatically drop encumbrances on EXPIRATION_DATE

**Comment by campmlc** [comment]
I would not want to drop Encumbering_agent_ID - knowing who set up the
encumbrance is important in figuring out why it is there and whether it is
still valid.  I have no problem with the other suggestions.  I like the
idea of email notifications.
Mariel

**Comment by dustymc** [comment]
Agreed, as long as we view ENCUMBERING_AGENT_ID as "who set up the encumbrance" rather than "who can revoke the encumbrance" (my current viewpoint, for whatever probably-incorrect reason).

Suggestions for a maximum EXPIRATION_DATE? Probably should be somehow vaguely related to the "normal" length of a research project (not that there's any such thing). My "5 years" suggestion now seems excessive (although I still like it better than "whenever one of the very few people who can now see the encumbered specimen realizes that the research project for which the encumbrance was created ended decades ago AND decides to do something about it...").

Somewhat related, why are we habitually masking entire records for "research purposes"? I believe many/most researchers would welcome complementary or unrelated usage - eg, if "AgentA" is working on genetics, why not simply "restrict usage" encumber, perhaps allowing AgentB to borrow other parts of the specimen for morphometric analyses (or whatever AgentA is comfortable with)? Is that a reasonable position, and if so how can we encourage it?

**Comment by campmlc** [comment]
I would suggest that we add links to the documentation for how to set up
encumbrances and the code table definitions for encumbrance actions to the
encumbrance page, to help people set them up correctly.

I think 5 years is reasonable before an encumbrance expires. But perhaps a
yearly email can be sent out informing collections of their existing
encumbrances, should they choose to end them earlier or extend them.

**Comment by dustymc** [comment]
Lacking timely objections, I intend to proceed with this as follows:

- expiration date becomes NOT NULL, with a maximum value of (now + 5 years)
- expiration event is removed, with existing data merged into remarks
- encumbering agent documentation is updated to "agent requesting the encumbrance; final authority to remove encumbrances rests with the collection"
- Email notifications are sent to collection contacts on...
-- expiration date minus multiples of 365 days
-- expiration date minus 180 days
-- expiration date minus 90 days
-- expiration date minus 30 days
-- expiration date

- expiration date becomes actionable; expired encumbrances will not restrict access to data.

**Comment by mkoo** [comment]
I'm fine with those actions. But on a more policy level, what about
encumbrances on things like locality for sensitive specimens (either for
conservation purposes or other?)--those may be in perpetuity. So I assume
can be continually renewed so the emailed reminders will be key to making
sure the right ones get renewed every 5 years, right?! (I just dont want to
assume the 5 years is an absolute limit)

**Comment by dustymc** [comment]
>those may be in perpetuity...

I don't think they (always, perhaps even ever) are, and that's the point of this. "SuperRareSubspecies" is declared sensitive, we go encumber everything, meanwhile it turns out that SuperRareSubspecies was never a valid classification, our specimens were misidentified, the habitat is now WWF corporate headquarters parking lot, the species has recovered, or WHATEVER. In the current model, there's no reason to ever review anything, the people who might care may not be able to see anything so won't remind you, etc., and the encumbrance lives on forever (while hiding critical data which could have been used to prevent that parking lot, had it only been available...). 

>5 years is an absolute limit

5 years is always from "now." This will be implemented as a before-trigger looking at eg "expiration_date <= add_months(sysdate,60)." There's no absolute maximum, and no intent to restrict what can be done with Encumbrances. Maybe SuperRareSubspecies will hang on for millennia, thanks only to your encumbrance. (I would like to review the documentation with anyone who wants to participate and clarify what SHOULD be done with Encumbrances.) This is intended only to encourage periodic review and discourage encumbrances which are no longer necessary or useful.

**Comment by dustymc** [comment]
for post-7.3.1 release:

- expiration_event is gone; data are in remarks
- everything that had no expiration date was updated to expire 5 years from NOW()
- everything that had a far-future expiration date was updated to expire 5 years from NOW()
- expiration_date is NOT NULL and must be less than 5 years from NOW()
- manage encumbrance form got a facelift
- the 1000-record batch-manage restriction removed
- email notifications are sent to collection contacts 0,6,12,24,36,48 months before an encumbrance expires
- expired encumbrances are visible only in the manage encumbrance form, and have no effect on specimens

**Comment by amgunderson** [comment]
Can encumbrances be easily renewed even after they expire? Is it just a matter of changing the expiration date to 5 years from now, no finding of specimens again required?

**Comment by dustymc** [comment]
Yup, just change the expiration date.

**Comment by campmlc** [comment]
Is it possible to keep an encumbrance history of what specimens have been encumbered by whom over time, even if they are no longer encumbered? Or, is it possible to keep track of what specimens have been attached to an encumbrance, even once they have been removed or the encumbrance expires?

**Comment by campmlc** [comment]
From A Johnson:
I'm surprised to learn that encumbrances now seem to only be good for
five years in Arctos. Could we please make it not be the default to
rescind encumbrances apparently without any notification as happened
with nearly all of ours on 1 March 2021? I have changed back the ones I
know I want masked or am not sure about but something happened on March
1 last year that I wasn't aware of.

I don't mind receiving 5 year reminders or something but removing
encumbrances without collection staff knowing is not appropriate, nor is
having the default be that they expire and are unmasked.

**Comment by campmlc** [comment]
I have experienced the same thing and agree with Andy that the default action after 5 years should not be to remove the encumbrance, but to keep it in place pending operator action. The consequences of revealing data that we have committed to keep secure, e.g. for researchers or govt agencies, are far more serious than the consequences of keeping stuff hidden that should be made public. People don't always get critical emails - I know I miss them - and we shouldn't put secure data at risk through missing an email notification.

**Comment by Jegelewicz** [comment]
Maybe this should be something you see when you log in and have manage collection in addition to an email when it gets down to the 90 and 30 day notifications?
